First Structure of the bootstrap project.

SigmaSam opened the pull request:
Build a recreation of the Newsweeks website using mostly using bootstrap classes.
Features
-Responsiveness for all screens.

SigmaSam committed:
Add boostrap framework.

SigmaSam committed:
Navbar for small screens.

SigmaSam committed:
Alpha stage website.

completed for first revision.

SigmaSam committed:
Updated .stickler.yml

SigmaSam committed:
Updated linters.

SigmaSam committed:
Errase local bootstrap folder

SigmaSam committed:
Fixed Linters errors

SigmaSam committed:
Another css error.

SigmaSam committed:
Centered the footer in the large resolution.

SigmaSam committed:
Added some screenshots.

SigmaSam committed:
Update README.md

SigmaSam committed:
Update README.md

SigmaSam committed:
Removed empty spaces.

SigmaSam committed:
Merge branch 'alpha' of https://github.com/SigmaSam/Using-Bootstrap into alpha

SigmaSam committed:
Removed conflit with linters.

SigmaSam commented:
Build a recreation of the Newsweeks website using mostly using bootstrap classes.
Features
-navigation bars hiding or showing according to the size of the user screen.
-Responsiveness for all screen sizes.
-Full use of bootstrap classes.

elmejdki commented:
## Pull Request under review by _@Zakariae El Mejdki_

elmejdki commented on README.md:
```diff
@@ -3,7 +3,9 @@ This project requests students to build a replica of the news site Newsweek.com
+
  ![screenshot](Images/screenshot3.png)
+
  ![screenshot](Images/screenshot2.png)
```
Fix the links in the images it's not working maybe you are not using an upper case 'i' letter in images.

elmejdki reviewed (changes requested):
# Good job so far @SigmaSam :clap: :clap: :clap:, your version looks really good :tophat:, but you need to do some changes in order to make your project approved.
### status: require changes.

- Fix the markdown to use a link you need to remove the space between ] and (.
![image](https://user-images.githubusercontent.com/34653764/84454808-857c4b80-ac53-11ea-9537-4e31a3daaab0.png)

- **[OPTIONAL]** add your LinkedIn and Twitter profile links also in your readme.

- Please add the Current date on the left top corner of your navbar as it is on the original page.
![image](https://user-images.githubusercontent.com/34653764/84455196-99747d00-ac54-11ea-9e55-19db7abe4669.png)

- your navbar menu breaks down on small screen size please fix it.
### Desktop view:
#### Your version:
:point_down:     :point_down:      :point_down:
![image](https://user-images.githubusercontent.com/34653764/84455383-21f31d80-ac55-11ea-8db4-b67de4b6f639.png)

#### Original page:
:point_down:       :point_down:       :point_down: 
![image](https://user-images.githubusercontent.com/34653764/84455421-39caa180-ac55-11ea-984a-96627c6f3f5c.png)

### Mobile view:
#### your version:
:point_down:     :point_down:      :point_down:
![image](https://user-images.githubusercontent.com/34653764/84455510-6da5c700-ac55-11ea-96b2-ffde9c3ee53d.png)

##### Original page:
![image](https://user-images.githubusercontent.com/34653764/84455556-83b38780-ac55-11ea-83ab-c91daea01dc2.png)

- you have a section the disappears in the top story section on tablet view:
![image](https://user-images.githubusercontent.com/34653764/84456390-c1191480-ac57-11ea-8d1c-88c423a3c3e5.png)

- Please fix the width of these two images on mobile version somehow they are not appearing as they should:
![image](https://user-images.githubusercontent.com/34653764/84456772-c2970c80-ac58-11ea-83dd-c24fa3e24990.png)

- Please fix this section on mobile view make it similar to the original one:
##### your version:
![image](https://user-images.githubusercontent.com/34653764/84456886-16a1f100-ac59-11ea-835c-ae103681d057.png)


##### original page:
![image](https://user-images.githubusercontent.com/34653764/84456904-228db300-ac59-11ea-981d-2616eb7b6c07.png)

- Please fix the newsletter and subscribe section to look similar to the original, on mobile and desktop view:
#### Desktop view:
##### Your version:
![image](https://user-images.githubusercontent.com/34653764/84457153-d55e1100-ac59-11ea-8b15-0b9c7e0361d1.png)

##### Original page:
![image](https://user-images.githubusercontent.com/34653764/84457171-de4ee280-ac59-11ea-8805-ebf838d9b6ee.png)

#### Mobile view:
##### Your version:
![image](https://user-images.githubusercontent.com/34653764/84457095-af387100-ac59-11ea-979c-e295470cdac0.png)

##### Original page:
![image](https://user-images.githubusercontent.com/34653764/84457075-a182eb80-ac59-11ea-8394-941a1415bfb0.png)

### please fix all the required changes before you request another code review.
_If you have any questions don't hesitate to contact me on slack_

SigmaSam committed:
Added the recomended fixes.

SigmaSam committed:
Filtered most sticklers errors.

SigmaSam commented:
Fixed the reviewd issues.

HADDADSOHAIB reviewed (changes requested):
### Status: Change Required :recycle: 

### Hello @SigmaSam   :hand: 

### Good job your project is getting better and better. :+1: 

### Unfortunately, there are still some changes to get the project approved:
- Please fix the conflict:
![Screenshot from 2020-06-15 12-11-41](https://user-images.githubusercontent.com/26064042/84651303-98b44300-af01-11ea-94d2-9f0964bfa05f.png)
- Please fix the images on README file:
![Screenshot from 2020-06-15 12-20-31](https://user-images.githubusercontent.com/26064042/84651950-af0ece80-af02-11ea-8d71-507d76a06bbe.png)
- I think this part of the mobile view still needs some improvement:
your project:
![Screenshot from 2020-06-15 12-26-57](https://user-images.githubusercontent.com/26064042/84652592-be424c00-af03-11ea-88d9-c8368d4d9edd.png)
Original:
![Screenshot from 2020-06-15 12-27-31](https://user-images.githubusercontent.com/26064042/84652605-c601f080-af03-11ea-8e85-fbc73259915d.png)
Think of reducing the margins and increasing the size of small images a little bit.
- The newsletter that you implemented is different then the original:
desktop, your project:
![Screenshot from 2020-06-15 12-32-06](https://user-images.githubusercontent.com/26064042/84652864-40327500-af04-11ea-83cc-b68ee9fc67e7.png)
desktop, original:
![Screenshot from 2020-06-15 12-32-31](https://user-images.githubusercontent.com/26064042/84652891-4e809100-af04-11ea-992e-0c113427b7aa.png)
mobile, your project:
![Screenshot from 2020-06-15 12-33-18](https://user-images.githubusercontent.com/26064042/84652965-6a843280-af04-11ea-9a0f-b3706a7aea9d.png)
mobile, original:
![Screenshot from 2020-06-15 12-33-55](https://user-images.githubusercontent.com/26064042/84653017-81c32000-af04-11ea-99f6-eb56d5a40060.png)
### Once you are done with the changes, ask for another code review.

SigmaSam committed:
Merge branch 'master' into alpha

SigmaSam committed:
Update README.md

SigmaSam committed:
Update README.md

SigmaSam committed:
Changed Images names.

SigmaSam committed:
Fixed issues in the review.

SigmaSam committed:
Fixed linters errors.

SigmaSam commented:
-Fixed conflict in the README.MD file.
-Fixed image size on smaller screens.
-Add image to newsletter in desktop screens.

SigmaSam committed:
Removed some syntax errors.

SigmaSam committed:
Revalidated with linters.

SigmaSam commented:
-Fixed conflict in the README.MD file.
-Fixed image size on smaller screens.
-Add image to newsletter in desktop screens.

HADDADSOHAIB reviewed (changes requested):
### Status: Change Required :recycle: 

### Hello @SigmaSam   :hand: 

### Thanks for fixing the readme file, but the other issues are not fixed. :+1: 
- As you can see here this part need a lot of improvement, focus of the margins or paddings, they are the ones that cause this, also the size of the font on mobile view should be smaller then the desktop view:
your project:
![Screenshot from 2020-06-15 14-32-02](https://user-images.githubusercontent.com/26064042/84663462-22214080-af15-11ea-9936-30dd5e2cea9f.png)

Original:
![Screenshot from 2020-06-15 12-27-31](https://user-images.githubusercontent.com/26064042/84652605-c601f080-af03-11ea-8e85-fbc73259915d.png)
Also, the horizontal scrolling should not be there, I am sorry I did not signal that before, all you need to do is to add this to css file:
`body {
    overflow-x: hidden;
}`
- The newsletter still need improvements:
mobile, your project:
![Screenshot from 2020-06-15 12-33-18](https://user-images.githubusercontent.com/26064042/84652965-6a843280-af04-11ea-9a0f-b3706a7aea9d.png)
mobile, original:
![Screenshot from 2020-06-15 12-33-55](https://user-images.githubusercontent.com/26064042/84653017-81c32000-af04-11ea-99f6-eb56d5a40060.png)
### Once you are done with the changes, ask for another code review.

SigmaSam committed:
Set a minimun size for images.

SigmaSam committed:
Fixed Linters errors and validated the code.

SigmaSam commented:
-Set a fixed image size.
-Set the subscribe section as a column in mobile resolutions.

addod19 reviewed (changes requested):
## Status: Require changes :recycle: :traffic_light: 

Hi @SigmaSam, great work fixing some of the comments made by the previous TSE :+1: However, there were some comments that were not addressed. I would like to highlight them so they get fixed. Kindly find comments below:
- ```Main navigation``` our main navigation needs to be fixed
![mainN](https://user-images.githubusercontent.com/46388113/84695349-3af01d00-af3a-11ea-8ddf-fb8888690bf1.png)
- Kindly fix the issues mentioned by the previous TSE's about the Subscribe and Newsletter section. It looks like the same from the previous TSE's comments.

Kindly make these changes and submit another code review :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

SigmaSam committed:
Fixed reviewed issues.

SigmaSam commented:
-Fixed the size of the font of the second navbar.

I honestly can't see the issue with the subscribe/newsletter here is a screen capture of how it looks in my screen, maybe is some issue with the browsers so I really don't know where to tackle it.

![image](https://user-images.githubusercontent.com/57440278/84701607-ad0d3580-af2b-11ea-8d32-fe1023ae5f42.png)

addod19 reviewed (changes requested):
## Status: Require changes :recycle: :traffic_light: 

Hi @SigmaSam, great work, and patience :+1: The following is a screenshot of the clone, source, and requirements:
![real1](https://user-images.githubusercontent.com/46388113/84716539-e8772680-af62-11ea-890b-0aebec05e94d.png)

![real](https://user-images.githubusercontent.com/46388113/84716519-db5a3780-af62-11ea-9650-130465f94f98.png)
![real2](https://user-images.githubusercontent.com/46388113/84716556-f9279c80-af62-11ea-9b07-a4185b2e12fd.png)
 
Per this requirement, we are advised to work to fix it. A quick fix will be to set a navContainer width: 100%; display: grid; grid-template-columns: repeat(no of list items, 1fr) or contact me if you may need assistance with this

Kindly submit a code review when done :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

SigmaSam committed:
Expanded th width of the navbar.

SigmaSam commented:
-Expanded the witdth of the second navigation bar as asked.

andresporras3423 commented:
### **Pull Request Under Review By Oscar Russi (TSE)**

andresporras3423 commented on styles.css:
```diff
@@ -0,0 +1,578 @@
+}
+
+.let__button {
+  background-color: #f72210 !important;
```
Don't overuse important

andresporras3423 commented on styles.css:
```diff
@@ -0,0 +1,578 @@
+  display: none;
+}
+
+#see_two a {
```
Don't overuse ID Selectors

andresporras3423 commented on styles.css:
```diff
@@ -0,0 +1,578 @@
+    display: none;
+  }
+
+  #more {
```
Don't overuse ID selectors

andresporras3423 reviewed (changes requested):
### Status: Changes Required :recycle:
**@SigmaSam** 
Good job with your page, it really looks like the original :muscle:  
Your page pass most of the requirements  :clap: :clap: :clap:  
There are just a few details to improve in your CSS code before you can go ahead:
- [x] Don't use important
- [x] Don't overuse ID Selectors

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras
(for any question you can contact me in slack as @oscarporras02)

SigmaSam committed:
Removed IDs and Important.

SigmaSam commented:
-Removed ID's.
-Removed !important.

tirthajyoti-ghosh reviewed (changes requested):
# Status: Changes required 🚨

## Great job implementing the changes 👏🙌!

### But we still need to work on some issues.

- The mobile version has extra space on the right side. The content should be full width in mobile version.

![image](https://user-images.githubusercontent.com/57726348/84817644-f09e9700-b032-11ea-9f4f-148629e2b7f8.png)

- In `the debate` section, the images are too large, in all screen sizes.

![image](https://user-images.githubusercontent.com/57726348/84817936-5d199600-b033-11ea-8a8d-be404821fc2d.png)

![image](https://user-images.githubusercontent.com/57726348/84817978-69055800-b033-11ea-985f-83a60f47bb54.png)

![image](https://user-images.githubusercontent.com/57726348/84818020-7589b080-b033-11ea-82db-49361eeb508c.png)

- This part of the navbar needs fixing.

![Annotation 2020-06-17 004426](https://user-images.githubusercontent.com/57726348/84818213-c8636800-b033-11ea-8f13-e1244fb15f2f.jpg)


Apply these changes and request another code review.

Happy coding 👨‍💻

SigmaSam committed:
Reduced the size of "the debete"

SigmaSam commented:
-Reduced the size of the debate container
-Fixed the size of the corner decoration of the second navigation bar.

andresporras3423 commented:
### **Pull Request Under Review By Oscar Russi (TSE)**

andresporras3423 commented on styles.css:
```diff
@@ -0,0 +1,586 @@
+  padding-bottom: 1rem;
+}
+
+.debate__art {
+  color: black;
+  font-size: 1rem;
+  font-weight: bold;
+}
```
This selector has the same styles as the selector on line 249. Group selectors to minimize code in your CSS file.

andresporras3423 commented on styles.css:
```diff
@@ -0,0 +1,586 @@
+  object-fit: contain;
+}
+
+.debater__one img {
+  object-fit: scale-down;
+  max-height: 90px;
+}
+
+.debater__two img {
+  object-fit: scale-down;
+  max-height: 90px;
+}
```
Don't repeat yourself. Group selectors to minimize code in your CSS file.

andresporras3423 commented on styles.css:
```diff
@@ -0,0 +1,586 @@
+  font-weight: bold;
+}
+
+.debate__art:hover {
+  text-decoration: none;
+  color: red;
+}
```
This selector has the same styles as the selector on line 255. Group selectors to minimize code in your CSS file.

andresporras3423 commented on styles.css:
```diff
@@ -0,0 +1,586 @@
+}
+
+@media (max-width: 768px) {
+  .hero {
+    display: none;
+  }
+
+  .more {
+    display: none;
+  }
```
Don't repeat yourself. Group selectors to minimize code in your CSS file.

andresporras3423 commented on styles.css:
```diff
@@ -0,0 +1,586 @@
+    margin-right: -8rem;
+  }
+
+  .priv {
+    margin-bottom: -5rem;
+  }
+
+  .in_story {
+    margin-top: -5rem;
+  }
```
Group selectors to minimize code in your CSS file.

andresporras3423 reviewed (changes requested):
### Status: Changes Required :recycle:
**Hello @SigmaSam** 
Good job so far  :muscle:  
I can see you have made a lot of improvements on your page :clap: :clap: :clap:  
You have to improve some parts:
- [x] Don't repeat yourself, in order to minimize code you can group selectors with the same style definition (in your CSS files).
- [x]  Check the navbar for missing details, compare the original:

![](https://i.imgur.com/0u3vlZLl.png)

With your clone:

![](https://i.imgur.com/u2lQ4Ifl.png%22)

Your page shows the hamburger icon even in the large screen, the hamburger icon should only appear in medium or smaller screen size version of the page. The search icon should be inside the input, not outside. Center the links of the navbar.

Now, let's check the footer, the original:

![](https://i.imgur.com/4c61YaYl.png)

And your footer:

![](https://i.imgur.com/jN7DGsNl.png)

Use all the width of the page for the link in your footer. The text at the right side of the page icon should keep in one line. The newsletter title should be above the image, not on the right side.

Check the "in the magazine" section.  It looks like this in your page:

![](https://i.imgur.com/rv2oZqjl.png)

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

SigmaSam committed:
Cleaned the code for repeating classes.

SigmaSam commented:
- Cleaned the code from unnecessary classes.
- Inserted glass icon in the navigation search box.
- Fixed word warp in some parts of the general document.
- Widened the footer for bigger resolutions.

peterrobert reviewed (changes requested):
### CHANGES REQUIRED :recycle: :stop_sign: :mechanic: 
**HELLO** @SigmaSam :wave: 
**GREAT PROJECT** You have made the changes required by the previous TSE :+1: 
However, before we approve this project you need to do the following changes:

- [ ] There need to be some space on the left and right on your webpage

**original**
![right_left_original](https://user-images.githubusercontent.com/32361963/84871143-d4c5e000-b088-11ea-8064-bdf5a06a25fc.jpg)

- [ ] Your menu items are being hidden in the medium view. Kindly improve on that

**Yours**
![menu_items_being_hidden](https://user-images.githubusercontent.com/32361963/84871272-0343bb00-b089-11ea-823c-6420e39ec211.jpg)

- [ ] At mobile view your webpage is losing responsiveness on the menu
**yours**
![menu_looses_repo_mobile](https://user-images.githubusercontent.com/32361963/84871424-2f5f3c00-b089-11ea-9e32-4d9d2ddbed8f.jpg)

**original**
![menu-looses_repo_mobile_original](https://user-images.githubusercontent.com/32361963/84871449-37b77700-b089-11ea-9059-a760f117ed0c.jpg)

**kindly improve on this before you send another code review** :smile: 

If you have any question about the review kindly contact me via slack on @peterrobert

### HAPPY CODING  :muscle:

SigmaSam committed:
Added Padding.

SigmaSam committed:
Add a default font besides font awesome.

SigmaSam commented:
- Add the horizontal padding.
- Add no wrapping setting to the small screen navigation bar.

peterrobert reviewed (changes requested):
### CHANGES REQUIRED :recycle: 
HEY @SigmaSam :wave:  
Great work implementing the requested changes :clap:  However there are a few more things you need to check.

- [ ] Kindly, You have not yet improved on your menu bar. You need to make sure it looks like the clone.
**original**
![search_bar_original](https://user-images.githubusercontent.com/32361963/84875436-a5b26d00-b08e-11ea-9732-878950165d45.jpg)
**yours**
![search_bar](https://user-images.githubusercontent.com/32361963/84875609-ddb9b000-b08e-11ea-8b8e-f40d5e72bb66.jpg)

- [ ] Also you need to make sure the menu bar is responsive. That is it moves with the container. This is because when you reduce the screen size the search bar is being hidden

**yours**
![search_disapper](https://user-images.githubusercontent.com/32361963/84875776-18bbe380-b08f-11ea-86b4-47a49c109f0a.jpg)

- [ ] Great job applying the changes that were required on the mobile menu. However, the hamburger icon is moving off the container  
**hint: make the navigation move with the container. This will increase responsiveness**

**yours**
![button_off_container](https://user-images.githubusercontent.com/32361963/84876045-70f2e580-b08f-11ea-94f4-22008088a143.jpg)

**Kindly make those changes before you ask for another review** :smile: 

if you have any question concerning the review kindly contact me via slack @peterrobert 

### HAPPY CODING :muscle:

SigmaSam committed:
resized second navigation bar.

SigmaSam committed:
Font Issues

SigmaSam commented:
- Resized second navigation bar.
- Set the navigation bar in smaller resolutions as a wraping one.

HADDADSOHAIB reviewed (approved):
### Status: Project Approved :+1:  

### Hello @SigmaSam   :hand: 

### Good job, you did great job with the project :+1: 

### You can merge this branch into master.

### There are some additional requirements that are **optional**:
- [Optional] You should maintain a professional github repository in all of your projects, that includes removing unnecessary files.
![Screenshot from 2020-06-17 11-47-48](https://user-images.githubusercontent.com/26064042/84889337-a3511280-b090-11ea-83e3-d7cab3a0e01c.png)
![Screenshot from 2020-06-17 11-47-40](https://user-images.githubusercontent.com/26064042/84889358-a8ae5d00-b090-11ea-83db-5fd4cb533567.png)
![Screenshot from 2020-06-17 11-47-34](https://user-images.githubusercontent.com/26064042/84889375-b06e0180-b090-11ea-955b-92d44e6bff4d.png)
